scripts/tests/yamltests/fixes.py

```diff
@@ -0,0 +1,116 @@
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an 'AS IS' BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import re
+import binascii
+
+'''Fixes certain value formats known to exist in YAML tests
+
+Some of the YAML test files contains some values that has been crafted to avoid some limitations
+of the original JavaScript parser and the C++ generated bits. This file contains functions to
+changes and convert those YAML values them back to something agnostic.
+'''
+
+
+def try_apply_yaml_cpp_longlong_limitation_fix(value):
+    '''Fix a known nasty hack in order that was used to work around compiler issue for C++.
+
+    When -9223372036854775808 was provided compiler would give a warning saying it is not a valid
+    way to write a long long in C++.
+    '''
+    if value == "-9223372036854775807LL - 1":
+        value = -9223372036854775808
+    return value
+
+
+def try_apply_yaml_unrepresentable_integer_for_javascript_fixes(value):
+    '''Fix up large integers that are represented within a string.
+
+    JavaScript can not represent integers bigger than 9007199254740991. But some of the test may
+    uses values that are bigger than this. The current way to workaround this limitation has
+    been to write those numbers as strings by encapsulating them in "".
+    '''
+    if type(value) is str:
+        value = int(value)
+    return value
+
+
+def try_apply_yaml_float_written_as_strings(value):
+    '''Fix up floats that are represented within a string.'''
+    if type(value) is str:
+        value = float(value)
+    return value
+
+
+# TODO(thampson) This method is a clone of the method in
+# src/controller/python/chip/yaml/format_converter.py and should eventually be removed in that file.
+def convert_yaml_octet_string_to_bytes(s: str) -> bytes:
+    '''Convert YAML octet string body to bytes.
+
+    This handles any c-style hex escapes (e.g. \x5a) and hex: prefix
+    '''
+    # Step 1: handle explicit "hex:" prefix
+    if s.startswith('hex:'):
+        return binascii.unhexlify(s[4:])
+
+    # Step 2: convert non-hex-prefixed to bytes
+    # TODO(#23669): This does not properly support utf8 octet strings. We mimic
+    # javascript codegen behavior. Behavior of javascript is:
+    #   * Octet string character >= u+0200 errors out.
+    #   * Any character greater than 0xFF has the upper bytes chopped off.
+    as_bytes = [ord(c) for c in s]
+
+    if any([value > 0x200 for value in as_bytes]):
+        raise ValueError('Unsupported char in octet string %r' % as_bytes)
+    accumulated_hex = ''.join([f"{(v & 0xFF):02x}" for v in as_bytes])
+    return binascii.unhexlify(accumulated_hex)
+
+
+def try_add_yaml_support_for_scientific_notation_without_dot(loader):
+    regular_expression = re.compile(u'''^(?:
```

Comment:
@vivien-apple 

Can you please break this down a little more to something that looks like this
```
...
# Comment about what you are looking for here, and why
fixes_inifinity = '[-+]?\\.(?:inf|Inf|INF)'
# Comment about what you are looking for here, and why
fixes_nan = '\\.(?:nan|NaN|NAN)'

'|'.join([..., fixes_inifinity, fixes_nan])
```

Reply:
Will be resolved in a follow up PR